projects/chaos/src/chaos_config.c

```diff
@@ -1,6 +1,105 @@
 #include "chaos_config.h"
 
-static ChaosConfig s_config;
+#include <inttypes.h>
+
+#include "gpio_mcu.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// ##### Voltage Conversion #####
+// 12 Bit ADC from 0 V to 3 V
+#define CHAOS_CONFIG_VALUE_TO_MILLIVOLTS(val) ((val)*3000 / (((uint16_t)1 << 12) - 1))
```

Comment:
If you're using the built-in ADC, `adc_read_converted` has a built-in reference that we use for a more accurate value

Reply:
Ack. I'll make that refactoring in a separate PR as it involved modifying power path which I would rather not add into this PR.